### ISSUE TITLE
Postgresql backup job failed with password authentication failed

### ISSUE THREAD

**Issue by fangpenlin** [kubedb/project#424]
I noticed that my postgresql backup job failed, due to error like this

```
failed to take backup
pg_dumpall: could not connect to database "template1": FATAL:  password authentication failed for user "postgres"
```

Turns out it looks like, I had `POSTGRES_USER` value set in secret, and it will create a super user with password I set

https://github.com/kubedb/postgres/blob/9c4f2e64642c3b690bb14bacca40161e2b1b0aac/hack/docker/postgres/10.2/scripts/primary/start.sh#L38

But look at the postgresql backup script 

https://github.com/kubedb/postgres/blob/f0c54f28dafb0f0d702ab17bf5f2125fe3c49091/hack/docker/postgres-tools/10.2/postgres-tools.sh#L24

the username is hardwired as `postgres`. Also, look at the job spec, there's no `POSTGRES_USER` environment variable set from my secret.

I guess the correct behavior for backup job should be, getting the same pg user as set in the secret?

**Comment by the-redback** [comment]
- [ ] Update postgres catalog in chart and bash.

**Comment by buehler** [comment]
In version 0.13 rc0, the problem still persists.
I just downloaded the postgres-tools kubedb image and ran it locally. When doing locally (with the actual version provided) it works like a charm.

I did prepare the following:

1. `docker run --it --entrypoint "" kubedb/postgres-tools:11.2 bash`
2. `export DB_HOST=host.docker.internal` (i set up a port forwarding to the actual database)
3. `export POSTGRES_USER=...`
4. `export POSTGRES_PASSWORD=...`
5. `postgres-tools.sh backup`

This results in the backup beeing created in `/var/data` (but not uploaded, since I did not provide the GCS secret. But it gets created. When running the backup job on kubernetes, I run into the same `pg_dumpall: could not connect to database "template1": FATAL:  password authentication failed for user "postgres"` error as stated above.

Do you have any idea why there is a discrepancy between those two?

Regards

**Comment by buehler** [comment]
Actually I think I found something. When using 
```bash
PGPASSWORD=$(echo -e "$POSTGRES_PASSWORD" | tr -d '[:space:]') pg_dumpall -U $DB_USER -h $DB_HOST > dump.sql
```

it works.

It seems that some kind of whitespace is added when using the variable from a secret.

**Comment by buehler** [comment]
@the-redback ? :-)

**Comment by tamalsaha** [comment]
This has been fixed in KubeDB/Stash v2021.03.17 .
